## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 428896`

Logs: https://github.com/ethancedwards8/nixpkgs-review-gha/actions/runs/16707239381

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-428896-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/ljd91zz1xf9fh4na1cyc1iyj6l0d18ys-python3.12-python3-sipsimple-5.3.3.2 \
  /nix/store/qk7028aya9y1i2kx6s39lw76ncq7wfii-python3.12-python3-sipsimple-5.3.3.2-dist \
  /nix/store/dld2xvr8d7ly0amzmpgrdq5l1kqyv3lr-python3.13-python3-sipsimple-5.3.3.2 \
  /nix/store/bygiz49pp3198qlqjnvacmxjqqbmqb5a-python3.13-python3-sipsimple-5.3.3.2-dist
```
</details></li>
</ul></details>


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.sipsimple</li>
    <li>python312Packages.sipsimple.dist</li>
    <li>python313Packages.sipsimple</li>
    <li>python313Packages.sipsimple.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.sipsimple</li>
    <li>python312Packages.sipsimple.dist</li>
    <li>python313Packages.sipsimple</li>
    <li>python313Packages.sipsimple.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.sipsimple</summary>
<pre>                 from ../../zsrtp/zrtp/srtp/CryptoContextCtrl.h:29,
                 from ../../zsrtp/zrtp/srtp/CryptoContextCtrl.cpp:27:
../../zsrtp/zrtp/cryptcommon/skeinApi.h:17: warning: ignoring ‘#pragma clang diagnostic’ [-Wunknown-pragmas]
   17 | #pragma clang diagnostic push
../../zsrtp/zrtp/cryptcommon/skeinApi.h:18: warning: ignoring ‘#pragma ide diagnostic’ [-Wunknown-pragmas]
   18 | #pragma ide diagnostic ignored "OCUnusedGlobalDeclarationInspection"
../../zsrtp/zrtp/cryptcommon/skeinApi.h:247: warning: ignoring ‘#pragma clang diagnostic’ [-Wunknown-pragmas]
  247 | #pragma clang diagnostic pop
ar: creating ../../lib/libzsrtp-aarch64-unknown-linux-gnu.a
../../webrtc/src/webrtc//modules/audio_processing/aec/aec_core_sse2.c:15:10: fatal error: emmintrin.h: No such file or directory
   15 | #include <emmintrin.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[3]: *** [/build/source/build/temp.linux-aarch64-cpython-312/pjsip/build/rules.mak:163: output/libwebrtc-aarch64-unknown-linux-gnu/modules/audio_processing/aec/aec_core_sse2.o] Error 1
make[2]: *** [Makefile:89: libwebrtc-aarch64-unknown-linux-gnu.a] Error 2
make[1]: *** [Makefile:7: all] Error 1
make: *** [Makefile:14: all] Error 1


ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
<details>
<summary>python313Packages.sipsimple</summary>
<pre>                 from ../../zsrtp/zrtp/srtp/CryptoContextCtrl.h:29,
                 from ../../zsrtp/zrtp/srtp/CryptoContextCtrl.cpp:27:
../../zsrtp/zrtp/cryptcommon/skeinApi.h:17: warning: ignoring ‘#pragma clang diagnostic’ [-Wunknown-pragmas]
   17 | #pragma clang diagnostic push
../../zsrtp/zrtp/cryptcommon/skeinApi.h:18: warning: ignoring ‘#pragma ide diagnostic’ [-Wunknown-pragmas]
   18 | #pragma ide diagnostic ignored "OCUnusedGlobalDeclarationInspection"
../../zsrtp/zrtp/cryptcommon/skeinApi.h:247: warning: ignoring ‘#pragma clang diagnostic’ [-Wunknown-pragmas]
  247 | #pragma clang diagnostic pop
ar: creating ../../lib/libzsrtp-aarch64-unknown-linux-gnu.a
../../webrtc/src/webrtc//modules/audio_processing/aec/aec_core_sse2.c:15:10: fatal error: emmintrin.h: No such file or directory
   15 | #include <emmintrin.h>
      |          ^~~~~~~~~~~~~
compilation terminated.
make[3]: *** [/build/source/build/temp.linux-aarch64-cpython-313/pjsip/build/rules.mak:163: output/libwebrtc-aarch64-unknown-linux-gnu/modules/audio_processing/aec/aec_core_sse2.o] Error 1
make[2]: *** [Makefile:89: libwebrtc-aarch64-unknown-linux-gnu.a] Error 2
make[1]: *** [Makefile:7: all] Error 1
make: *** [Makefile:14: all] Error 1


ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
</details>


---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*


---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

--- a/pkgs/development/python-modules/sipsimple/default.nix
+++ b/pkgs/development/python-modules/sipsimple/default.nix
@@ -1,0 +1,136 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  fetchurl,
+  fetchpatch,
+  cython,
+  setuptools,
+  alsa-lib,
+  ffmpeg,
+  libopus,
+  libuuid,
+  libv4l,
+  libvpx,
+  opencore-amr,
+  openssl,
+  pkg-config,
+  sqlite,
+  x264,
+  python3-application,
+}:
+
+let
+  extDeps = {
+    pjsip = rec {
+      # Hardcoded in get_dependencies.sh, checked at buildtime
+      # need tarball specifically for buildscript to detect it
+      version = "2.10";
+      src = fetchurl {
+        url = "https://github.com/pjsip/pjproject/archive/${version}.tar.gz";
+        hash = "sha256-k2pMW5hgG1IyVGOjl93xGrQQbGp7BPjcfN03fvu1l94=";
+      };
+    };
+    zrtpcpp = rec {
+      # Hardcoded in get_dependencies.sh, NOT checked at buildtime
+      rev = "6b3cd8e6783642292bad0c21e3e5e5ce45ff3e03";
+      src = fetchFromGitHub {
+        owner = "wernerd";
+        repo = "ZRTPCPP";
+        inherit rev;
+        hash = "sha256-kJlGPVA+yfn7fuRjXU0p234VcZBAf1MU4gRKuPotfog=";
+      };
+    };
+  };
+in
+buildPythonPackage rec {
+  pname = "python3-sipsimple";
+  version = "5.3.3.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "AGProjects";
+    repo = "python3-sipsimple";
+    tag = "${version}-mac";
+    hash = "sha256-kDXVzLmgfXxm8phKrV7DvPuZ9O2iNFo1s6Lc0jcc/dM=";
+  };
+
+  patches = [
+    # Remove when version > 5.3.3.2-mac
+    (fetchpatch {
+      name = "0001-python3-sipsimple-port-to-cython-3.patch";
+      url = "https://github.com/AGProjects/python3-sipsimple/commit/ccbbbb0225b2417bdf6ae07da96bffe367e242be.patch";
+      hash = "sha256-MBiM9/yS5yX9QoZT7p76PI2rbBr22fZw6TAT4tw9iZk=";
+    })
+  ];
+
+  postPatch = ''
+    substituteInPlace get_dependencies.sh \
+      --replace-fail 'sudo apt' 'echo Skipping sudo apt'
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  build-system = [
+    cython
+    setuptools
+  ];
+
+  buildInputs = [
+    alsa-lib
+    ffmpeg
+    libopus
+    libuuid
+    libv4l
+    libvpx
+    opencore-amr
+    openssl
+    sqlite
+    x264
+  ];
+
+  dependencies = [
+    python3-application
+  ];
+
+  preConfigure = ''
+    ln -s ${passthru.extDeps.pjsip.src} deps/${passthru.extDeps.pjsip.version}.tar.gz
+    cp -r --no-preserve=all ${passthru.extDeps.zrtpcpp.src} deps/ZRTPCPP
+
+    bash ./get_dependencies.sh
+
+    # Fails to link some static libs due to missing -lc DSO. Just use the compiler frontend instead of raw ld.
+    substituteInPlace deps/pjsip/build/rules.mak \
+      --replace-fail '$(LD)' "$CC"
+
+    # Incompatible pointers (not const)
+    substituteInPlace deps/pjsip/pjmedia/src/pjmedia-codec/ffmpeg_vid_codecs.c \
+      --replace-fail '&payload,' '(const pj_uint8_t **)&payload,'
+  '';
+
+  # no upstream tests exist
+  doCheck = false;
+
+  pythonImportsCheck = [ "sipsimple" ];
+
+  passthru = {
+    inherit extDeps;
+  };
+
+  meta = {
+    description = "SIP SIMPLE SDK written in Python";
+    homepage = "https://sipsimpleclient.org/";
+    downloadPage = "https://github.com/AGProjects/python3-sipsimple";
+    license = lib.licenses.gpl3Plus;
+    teams = [ lib.teams.ngi ];
+    maintainers = [ lib.maintainers.ethancedwards8 ];
+    badPlatforms = [
+      # ../../webrtc/src/webrtc//modules/audio_processing/aec/aec_core_sse2.c:15:10: fatal error: emmintrin.h: No such file or directory
+      "aarch64-linux"
+    ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -16481,6 +16481,8 @@ self: super: with self; {
 
   siphashc = callPackage ../development/python-modules/siphashc { };
 
+  sipsimple = callPackage ../development/python-modules/sipsimple { };
+
   sipyco = callPackage ../development/python-modules/sipyco { };
 
   sirius = toPythonModule (

